ipynb/examples/utils/gem5_integration.ipynb

```diff
@@ -0,0 +1,342 @@
+    "Before creating the gem5 target, the inputs needed by gem5 should have been created (eg gem5 binary, kernel suitable for gem5, disk image, device tree blob, etc). For more information, see [GEM5 - Main Page](http://gem5.org/Main_Page)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "# Environment setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "# One initial cell for imports\n",
+    "import json\n",
+    "import logging\n",
+    "\n",
+    "from env import TestEnv"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "from conf import LisaLogging\n",
+    "LisaLogging.setup()\n",
+    "\n",
+    "logging.getLogger('ssh').setLevel(logging.DEBUG)\n",
+    "logging.getLogger('gem5-connection').setLevel(logging.DEBUG)\n",
+    "logging.getLogger('gem5').setLevel(logging.DEBUG)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true,
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "# Target configuration\n",
+    "The definitions below need to be changed to the paths pointing to the gem5 binaries on your development machine.\n",
+    "M5_PATH needs to be set in your environment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "conf = {\n",
+    "    # Only 'linux' is supported by gem5 for now\n",
+    "    # 'android' is a WIP\n",
+    "    \"platform\"    : 'linux',\n",
+    "    \n",
+    "    # Preload settings for a specific target\n",
+    "    \"board\"       : 'gem5',\n",
+    "\n",
+    "    # Host that will run the gem5 instance\n",
+    "    \"host\"        : \"hostname\",\n",
+    "    \n",
+    "    \"gem5_settings\" : {\n",
+    "        # System to simulate\n",
+    "        \"system\" : {\n",
+    "            # Platform description\n",
+    "            \"platform\" : {\n",
+    "                \"description\" : \"/home/user/platform_juno/juno.py\",\n",
+    "                \"args\" : [\n",
+    "                    # Resume simulation from a previous checkpoint\n",
+    "                    # Checkpoint must be taken before Virtio folders are mounted\n",
+    "                    \"--checkpoint-indir /home/user/gem5/platform_juno/checkpoints\",\n",
+    "                    \"--checkpoint-resume 1\",\n",
+    "                ]\n",
+    "            },\n",
+    "            # Kernel compiled for gem5 with Virtio flags\n",
+    "            \"kernel\"   : \"/home/user/Work/gem5/platform_juno/vmlinux\",\n",
```

Comment:
Do you mean to define  base folder in the notebook or in the actual gem5 API? If the latter then we need to be really careful to ensure we don't place annoying restrictions on the relative locations of the different binaries..